src/UserAdmin/src/Smart.FA.Catalog.Core/Extensions/CollectionExtensions.cs

```diff
@@ -0,0 +1,23 @@
+namespace System.Collections;
+
+public static class CollectionExtensions
+{
+    /// <summary>
+    /// Determines if there is an element of type <typeparam name="T"> </typeparam> inside a <see cref="IEnumerable" />.
+    /// </summary>
+    /// <typeparam name="T">Actual type to look for</typeparam>
+    /// <param name="source">Source collection.</param>
+    /// <returns>True if there is an element whose type is <typeparam name="T"/></returns>
+    public static bool AnyOfType<T>(this IEnumerable source)
+    {
+        foreach (var obj in source)
+        {
+            if (obj is T)
+            {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
Using the `Any` method is cleaner and shorter
```suggestion
        return source.Any(object => object is T);
```

Reply:
For those kind of situation I prefer using the foreach over linq. 
It is just a matter of performance.
I even prefer using List.Count > 0 instead of any() in most situation but I know ppl prefer it more readable :).